src/pyflex/window_selector.py

```diff
@@ -688,10 +765,7 @@ def curtail_window_length(win):
             time_decay_right = self.config.min_period * self.config.c_4b / dt
             # Find all internal maxima.
             internal_maxima = self.peaks[
-                (self.peaks >= win.left) & (self.peaks <= win.right) &
-                (self.peaks != win.center)]
-            if len(internal_maxima) < 2:
-                return win
+                (self.peaks >= win.left) & (self.peaks <= win.right) ]
```

Comment:
What is the motivation behind this? You don't want to add the current peak to the list of peaks to feed into the curtailing and it does not work with less then two additional peaks.